### ISSUE TITLE
NewAdaptiveGrid with widget width ratio.

### ISSUE THREAD

**Issue by ayush-raj8** [fyne-io/fyne#3738]
### Checklist

- [X] I have searched the [issue tracker](https://github.com/fyne-io/fyne/issues) for open issues that relate to the same feature, before opening a new one.
- [X] This issue only relates to a single feature. I will open new issues for any other features.

### Is your feature request related to a problem?

As of now, if I put a widget list in NewAdaptiveGrid it renders them to be of equal size.

```
myWin.SetContent(container.NewVBox(
		container.NewAdaptiveGrid(3, uploadButton, downloadButton, locationButton),
		container.NewAdaptiveGrid(4, blank, pgBar, progressBar, blank),
		layout.NewSpacer(),
	),
	)
```
As a result I see this :

<img width="1512" alt="Screenshot 2023-03-16 at 11 55 55 AM" src="https://user-images.githubusercontent.com/52033949/225533530-0eca6e73-2ccd-4f8a-8883-bb1371ce9fd3.png">

### Is it possible to construct a solution with the existing API?

Yes maybe, by adding a new type.

### Describe the solution you'd like to see.

I would love to see progress bar Text on the left size occupying 3/10 of row row and the actual progress bar occupying 7/10 of the row width.

**Comment by andydotxyz** [comment]
Perhaps this is what you are looking for: https://github.com/fyne-io/fyne-x#responsive-layout

**Comment by ayush-raj8** [comment]
@andydotxyz yes one can implement a long custom code to do so but we should have an implementation besides [NewAdaptiveGrid](https://github.com/fyne-io/fyne/blob/master/layout/gridlayout.go) that helps control the relative sizes of widgets in a row. 
That may give UI like this with a high degree of encapsulation.
<img width="660" alt="Screenshot 2023-03-16 at 5 32 55 PM" src="https://user-images.githubusercontent.com/52033949/225612690-5b72975b-ba59-4471-a1d0-478da78fdc0f.png">

Intended code to result:
```
myWindow.SetContent(container.NewVBox(
		NewAdaptiveGridWithRatios([]float32{0.1, 0.4, 0.4, 0.1}, buttons[0], buttons[1], buttons[2], buttons[3]),
		NewAdaptiveGridWithRatios([]float32{0.2, 0.3, 0.1, 0.4}, buttons[4], buttons[5], buttons[6], buttons[7]),
		NewAdaptiveGridWithRatios([]float32{0.6, 0.1, 0.2, 0.1}, buttons[8], buttons[9], buttons[10], buttons[11]),
		NewAdaptiveGridWithRatios([]float32{0.1, 0.4, 0.4, 0.1}, buttons[12], buttons[13], buttons[14], buttons[15]),
		NewAdaptiveGridWithRatios([]float32{0.1, 0.9}, pgBar, progressBar),
	))

```

**Comment by andydotxyz** [comment]
@ayush-raj8 the progress bar example at the bottom of your demo screen is currently possible using `Border` container. That works on "min" and "the rest" sizing which is, more adaptable to smaller screens than "progress is 9* size of label".

Did you look at the link I proveded? I was not suggesting that everyone should manually code this, but that there is a container that handles much of what you are asking already existing: https://github.com/fyne-io/fyne-x#responsive-layout

**Comment by andydotxyz** [comment]
This has been implemented in Fyne-x

https://github.com/fyne-io/fyne-x/blob/master/layout/portion.go
